ProcessMaker/spark
# Add support for translations

Fixes #1617 

Uses default laravel translations, i18next libraries, and ProcessMaker/laravel-i18next to translate entire site. New translations added in the spark-package-translations package

Relies on
- https://github.com/ProcessMaker/vue-form-builder/pull/138
- https://github.com/ProcessMaker/modeler/pull/312
- https://github.com/ProcessMaker/spark-packages
- https://github.com/ProcessMaker/spark-package-translations

### Testing
Until spark-packages and spark-package-translations are released, you need to clone those packages locally in your bpm-plugins folder (so vagrant has access to them). Then edit spark's composer.json and add to "repositories"
```
    ,{
      "type": "vcs",
      "url": "https://github.com/ProcessMaker/spark-packages.git"
    },
    {
      "type": "vcs",
      "url": "https://github.com/ProcessMaker/spark-package-translations.git"
    }
```

  - Then, require them
     `composer require processmaker/spark-packages:dev-master`
     `composer require processmaker/spark-package-translations:dev-master`

  - Run the install scripts for both
    `php artisan spark-packages:install`
    `php artisan package:install spark-package-translations`

  - Edit .env and change the APP_ENV to es
  - run `npm run watch`
  - Edit the users profile and change language to es
  - load the app and verify all strings are 'yo soy espanol'
  - Change user language setting to fr
  - verify strings are changed to 'je suis francais'
